code/modules/clothing/suits/hoodies.dm

```diff
@@ -24,10 +25,14 @@
 	..()
 
 /obj/item/clothing/suit/storage/hooded/proc/RemoveHood()
-	icon_state = initial(icon_state)
-	item_state = initial(item_state)
-	suittoggled = 0
-
+	if(!lasticonstate)
+		suittoggled = FALSE
+	else
+		icon_state = "[lasticonstate]"
+		item_state = "[lasticonstate]"
```

Comment:
You can just do this:
`icon_state = lasticonstate`
We only need to use brackets if you need to append a string to it.